src/proto/uprotocol/uauthority.rs

```diff
@@ -0,0 +1,65 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::uprotocol::{Remote, UAuthority};
+
+/// Helper functions to deal with UAuthority::Remote structure
+impl UAuthority {
+    pub fn has_name(authority: &UAuthority) -> bool {
```

Comment:
It seems that all functions defined in `impl UAuthority` take as first parameter either `&UAuthority` or `&mut UAuthority`.
Why not taking `&self` and `&mut self` instead as first parameter?